### PR TITLE
Make "openssl rehash" work on VMS 8.3 and up

### PR DESCRIPTION
**Pull request by levitte**
A spelling error stopped this from working

### CODE REVIEW

**Review by richsalz — APPROVED** [review]

**Comment by levitte** [comment]
Hold ye 'orses, there's more.

**Comment by levitte** [comment]
There, *now* I'm done

File: apps/rehash.c [hunk 1]
```diff
@@ -267,6 +267,21 @@ static void str_free(char *s)
     OPENSSL_free(s);
 }
 
+static int slashy_dir_end_needed(const char *path)
```
Comment (richsalz) [inline apps/rehash.c]: if you invert the sense of the test, it looks simpler.
```
static int ends_with_dirsep(const char *path)
{
    if (*path != '\0')
        path += strlen(path) - 1;
#ifdef __VMS
    return *path == ']' || *path == '>' || *patch == ':';
... etc
```

**Comment by levitte** [comment]
(@richsalz, this needs re-confirming...  sorry)

**Review by richsalz — CHANGES_REQUESTED** [review]

File: apps/rehash.c (hunk 1 above)
Comment (levitte) [inline apps/rehash.c]: I'm not keen on those immediate returns, as both the Windows and VMS C libraries can handle Unix-style paths as well.  But other than that, sure.  Coming up.

**Comment by levitte** [comment]
Have a look now, @richsalz

**Review by richsalz — APPROVED** [review]

**Comment by levitte** [comment]
Merged

master: 5c80e2af3a7d8aa5129a1668c286c1464983e1ac
1.1.0: d8d3b66971dab2b54dd0946d85fd45a2470714a8

**Pull request closed without merging**